mesonbuild/modules/gnome.py

```diff
@@ -45,7 +45,11 @@ def compile_resources(self, state, args, kwargs):
         if not isinstance(source_dirs, list):
             source_dirs = [source_dirs]
 
-        ifile = args[1]
+        try:
+            ifile = args[1]
+        except IndexError:
```

Comment:
I know this is an outdated diff, but this is a pretty standard way of doing things in Python. As long as your `try:` contents are as small as possible, it's not really unclear and is fairly [Pythonic](https://docs.python.org/3/glossary.html#term-eafp).